### PR TITLE
sys-sage: new spack package

### PR DESCRIPTION
**Pull request by stepanvanecek**
Adding sys-sage to spack

https://github.com/stepanvanecek/sys-sage

### CODE REVIEW

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    version('master',  branch='master')
+    version('0.1.1-alpha.2', sha256='991a77cf37b061a911c8566fd4486f914de4f4c8cdf39112ec8a32903450c178')
+
+    depends_on('cmake@3.21:', type='build')
+    depends_on('libxml2@2.9.13')
+   
+   #def cmake_args(self):
+    #    args = []
+    #    return args
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sys-sage/package.py]: In general we discourage retaining comments unless there is a good reason provided.

Do you anticipate adding `cmake` arguments in the future?

File: var/spack/repos/builtin/packages/sys-sage/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    """A library for capturing hadrware topology and attributes of compute systems."""
+
+    homepage = "https://github.com/stepanvanecek/sys-sage"
+    url      = "https://github.com/stepanvanecek/sys-sage/archive/refs/tags/v0.1.1-alpha.2.tar.gz"
+    git      = "https://github.com/stepanvanecek/sys-sage.git"
+
+    maintainers = ['stepanvanecek']
+
+    version('master',  branch='master')
+    version('0.1.1-alpha.2', sha256='991a77cf37b061a911c8566fd4486f914de4f4c8cdf39112ec8a32903450c178')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/sys-sage/package.py:20]: Confirmed the sha256.

**Comment by tldahlgren** [comment]
@stepanvanecek For future reference, IME it is easier for us to review PRs when they are on a non-develop branch.

**Comment by tldahlgren** [comment]
I attempted `spack install sys-sage` on a linux machine at LLNL and I get the following error:

```
==> Error: Package 'libllvm' not found.
```

File: var/spack/repos/builtin/packages/sys-sage/package.py (hunk 1 above)
Comment (stepanvanecek) [reply var/spack/repos/builtin/packages/sys-sage/package.py]: I indeed plan to use `cmake` arguments later on but I can as well remove it for now and add it later once needed.

**Comment by stepanvanecek** [comment]
@tldahlgren thank you for the review

- Next time, I will open the PR from a feature branch

- I was not able to reproduce the `==> Error: Package 'libllvm' not found.` on any (3) machines I tried (but all used gcc). Do I need llvm to reproduce this? I added a `libllvm@12` dependency but am not sure whether it solved the problem you saw. If not, could you please tell me how to reproduce/debug the issue?

**Comment by tldahlgren** [comment]
> I attempted `spack install sys-sage` on a linux machine at LLNL and I get the following error:
> 
> ```
> ==> Error: Package 'libllvm' not found.
> ```

I believe this is a problem with my configuration so feel free to ignore.

**Comment by tldahlgren** [comment]
@stepanvanecek Is this still a DRAFT PR?  If not you can click `Ready for review` so the full set of checks are performed.

**Comment by stepanvanecek** [comment]
Great, thanks. Then I will remove the `libllvm@12` dependency, re-run the tests locally, and then mark as `Ready for review`.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**